packages/core/src/awsService/ec2/sshKeyPair.ts

```diff
@@ -26,14 +40,38 @@ export class SshKeyPair {
         if (result.exitCode !== 0) {
             throw new ToolkitError('ec2: Failed to generate ssh key', { details: { stdout: result.stdout } })
         }
+        await chmod(keyPath, 0o600)
     }
 
     public getPublicKeyPath(): string {
         return this.publicKeyPath
     }
 
+    public getPrivateKeyPath(): string {
+        return this.keyPath
+    }
+
     public async getPublicKey(): Promise<string> {
-        const contents = await fs.readFile(this.publicKeyPath, 'utf-8')
+        const contents = new TextDecoder().decode(await fs.readFile(this.publicKeyPath))
         return contents
     }
+
+    public async delete(): Promise<void> {
+        await fs.delete(this.publicKeyPath)
+        await fs.delete(this.keyPath)
```

Comment:
might be a good idea to delete the private key first, in case deleting the public key fails (which is less important)